### PR TITLE
bug(query) Fix invalid start/end parameters for PeriodicSamplesMapper for subqueries in MultiPartitionPlanner

### PR DESCRIPTION
**Pull request by kvpetrov**
… for subqueries in MultiPartitionPlanner

**Pull Request checklist**

- [ ] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [ ] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

**Current behavior :** (link exiting issues here : https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests)

**New behavior :**

**BREAKING CHANGES**

If this PR contains a breaking change, please describe the impact and migration
path for existing applications.
If not please remove this section.

Breaking changes may include:
- Any schema changes to any Cassandra tables
- The serialized format for Dataset and Column (see .toString methods)
- Over the wire formats for Akka messages / case classes
- Changes to the HTTP public API
- Changes to query parsing / PromQL parsing

**Other information**:

### CODE REVIEW

File: coordinator/src/test/scala/filodb.coordinator/queryplanner/MultiPartitionPlannerSpec.scala [hunk 1]
```diff
@@ -175,6 +175,105 @@ class MultiPartitionPlannerSpec extends AnyFunSpec with Matchers {
...
+        List(PartitionAssignment("local", "local-url", TimeRange(timeRange.startMs, timeRange.endMs)))
+    }
+
+    val engine = new MultiPartitionPlanner(partitionLocationProvider, localPlanner, "local", dataset, queryConfig)
+    val lp = Parser.queryRangeToLogicalPlan(
+      "test{job = \"app\"}[100m:10m]",
+      TimeStepParams(endSeconds, step, endSeconds)
+    )
+
+    val promQlQueryParams = PromQlQueryParams("test{job = \"app\"}", endSeconds, step, endSeconds)
```
Comment (TanviBhavsar) [inline coordinator/src/test/scala/filodb.coordinator/queryplanner/MultiPartitionPlannerSpec.scala:193]: This should have query "test{job = \"app\"}[100m:10m]" right?

File: coordinator/src/test/scala/filodb.coordinator/queryplanner/MultiPartitionPlannerSpec.scala [hunk 2]
```diff
@@ -175,6 +175,105 @@ class MultiPartitionPlannerSpec extends AnyFunSpec with Matchers {
...
```
Comment (TanviBhavsar) [inline coordinator/src/test/scala/filodb.coordinator/queryplanner/MultiPartitionPlannerSpec.scala:246]: Same as above. Query should be 'test{job = \"app\"}[100m:10m]'

**Review by TanviBhavsar — APPROVED** [review]
Looks good to me. Approved with minor comment

File: coordinator/src/test/scala/filodb.coordinator/queryplanner/MultiPartitionPlannerSpec.scala (hunk 1 above)
Comment (kvpetrov) [reply coordinator/src/test/scala/filodb.coordinator/queryplanner/MultiPartitionPlannerSpec.scala]: fixed

File: coordinator/src/test/scala/filodb.coordinator/queryplanner/MultiPartitionPlannerSpec.scala (hunk 2 above)
Comment (kvpetrov) [reply coordinator/src/test/scala/filodb.coordinator/queryplanner/MultiPartitionPlannerSpec.scala]: fixed

**Pull request merged**